### PR TITLE
fix(pattern): reduced heading margin on content-group

### PR DESCRIPTION
**Pull request by guilhermelMoraes**
### Related Ticket(s)

Content Group Simple renders item title too high above optional image #1890

### Description

as related in the ticket above, the h3 heading margin-bottom was bigger than expected, so I reduced it.  

### Changelog

**Changed**

- image-with-caption.scss - image-with-caption class margin-top;
- _content-group.scss - title class margin-bottom.

### CODE REVIEW

**Comment by jeffchew** [comment]
Adding designers to this review as well (@wonilsuhibm @larahanlon2 @oliviaflory )

**Comment by kennylam** [comment]
> I don't think this change is being applied in the right place.

@oliviaflory It's applied in the correct spot, no?

- [deploy preview](https://ibmdotcom-react.s3.us-south.cloud-object-storage.appdomain.cloud/deploy-previews/1975/index.html?path=/story/patterns-blocks-contentgroupsimple--default) above (Guilherme's fix):
image

- [ibmdotcom-react deploy](http://ibmdotcom-react.mybluemix.net/?path=/story/patterns-blocks-contentgroupsimple--default) (what Ke reported)
image

**Comment by oliviaflory** [comment]
Huh, I thought what Ke reported was the spacing when an image appears between the semibold heading and body copy sub section

Ke's screen shot:
Screen Shot 2020-04-07 at 9 36 18 AM

Our change looks like it's being applied to the heading of the overall section.

Screen Shot 2020-04-06 at 5 29 03 PM

**Comment by jeffchew** [comment]
@oliviaflory @guilhermelMoraes so where did we land on this, we should be making the change to ContentItem and not ContentGroup, correct?

**Comment by oliviaflory** [comment]
> @oliviaflory @guilhermelMoraes so where did we land on this, we should be making the change to ContentItem and not ContentGroup, correct?

I thought we had a meeting Roberta set up at 10:45 est today to discuss

**Comment by oliviaflory** [comment]
Reviewed this PR today and Roberta captured notes on what needs to be updated in #1890 to resolve the changes

**Comment by larahanlon2** [comment]
Just to add – we may need to revert the change that was made to the main heading to make sure the padding is correct.

Currently in experimental [ContentGroup ](https://ibmdotcom-react-experimental.s3.us-south.cloud-object-storage.appdomain.cloud/deploy-previews/1975/index.html?path=/story/patterns-sub-patterns-contentgroup--default)Storybook:
Screen Shot 2020-04-08 at 12 02 13 PM

Actual specs. Space between heading and image should be 32px:
Screen Shot 2020-04-08 at 12 05 31 PM

File: packages/styles/scss/components/image-with-caption/image-with-caption.scss [hunk 1]
```diff
@@ -9,13 +9,13 @@
 
 @mixin image-with-caption {
   .#{$prefix}--image-with-caption {
-    margin-top: $carbon--spacing-07;
+    margin-top: $carbon--spacing-06;
```
Comment (annawen1) [inline packages/styles/scss/components/image-with-caption/image-with-caption.scss:12]: did we need to change the margins for the `ImageWithCaption` component? I thought we were just changing the styling in `ContentItem`

Comment (guilhermelMoraes) [reply packages/styles/scss/components/image-with-caption/image-with-caption.scss:12]: I did it to match the match the specs that Lara mentioned above.

Comment (oliviaflory) [reply packages/styles/scss/components/image-with-caption/image-with-caption.scss:12]: @guilhermelMoraes I believe Lara was reminding us to remove the changes you had initially made to the `Content Group` `margin-top` when you first sent the PR. 

I was also under the impression we didn't need to change the `ImageWithCaption` component. However, the most recent visual specs for media is 32px above and below, so not sure if we want to fix that here or later

Comment (annawen1) [reply packages/styles/scss/components/image-with-caption/image-with-caption.scss:12]: if that's the case I would revert back to `margin-top: to $carbon--spacing-07`

Comment (jeffchew) [reply packages/styles/scss/components/image-with-caption/image-with-caption.scss:12]: Let's leave ImageWithCaption alone for now. There's work coming up with the Media manager that will be addressing padding there.

Comment (guilhermelMoraes) [reply packages/styles/scss/components/image-with-caption/image-with-caption.scss:12]: Ok got it. So I'll keep the changes only in `ContentItem` like we discussed previously.

**Pull request merged**